### PR TITLE
Remove boost test checks from SMTCheckerJSONTest

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

File: test/libsolidity/SMTCheckerJSONTest.cpp [hunk 1]
```diff
@@ -38,11 +38,14 @@ using namespace boost::unit_test;
...
 : SyntaxTest(_filename)
 {
-	BOOST_REQUIRE_MESSAGE(boost::algorithm::ends_with(_filename, ".sol"), "Invalid test contract file name: \"" + _filename + "\".");
+	if (!boost::algorithm::ends_with(_filename, ".sol"))
+		BOOST_THROW_EXCEPTION(runtime_error("Invalid test contract file name: \"" + _filename + "\"."));
 
 	string jsonFilename = _filename.substr(0, _filename.size() - 4) + ".json";
-	BOOST_CHECK(jsonParseFile(jsonFilename, m_smtResponses));
-	BOOST_CHECK(m_smtResponses.isObject());
+	if (!jsonParseFile(jsonFilename, m_smtResponses) ||
```
Comment (chriseth) [inline test/libsolidity/SMTCheckerJSONTest.cpp]: either all in one line or
```
if (
    !json... ||
    !m_stmt...
)
```

File: test/libsolidity/SMTCheckerJSONTest.cpp [hunk 2]
```diff
@@ -59,42 +62,59 @@ bool SMTCheckerTest::run(ostream& _stream, string const& _linePrefix, bool const
...
+			boost::algorithm::join(outHashes, ", ") +
+			" x " +
+			boost::algorithm::join(inHashes, ", ")
+		));
 
 	// Rerun the compiler with the provided hashed (2nd run)
 	input[auxInput] = m_smtResponses[auxInput];
 	Json::Value endResult = compiler.compile(input);
 
-	BOOST_CHECK(endResult.isMember("errors"));
```
Comment (chriseth) [inline test/libsolidity/SMTCheckerJSONTest.cpp:75]: Would be good to still have exceptions here.

File: test/libsolidity/SMTCheckerJSONTest.cpp [hunk 3]
```diff
@@ -59,42 +62,59 @@ bool SMTCheckerTest::run(ostream& _stream, string const& _linePrefix, bool const
...
-			error["type"].asString(),
-			error["message"].asString(),
-			start,
-			end
-		});
+		Json::Value const& errors = endResult["errors"];
+		for (auto const& error: errors)
+		{
+			if (!error.isMember("type") ||
+				!error["type"].isString()
```
Comment (chriseth) [inline test/libsolidity/SMTCheckerJSONTest.cpp:91]: style

File: test/libsolidity/SMTCheckerJSONTest.cpp [hunk 4]
```diff
@@ -59,42 +62,59 @@ bool SMTCheckerTest::run(ostream& _stream, string const& _linePrefix, bool const
...
+			)
+				BOOST_THROW_EXCEPTION(runtime_error("Error must have a type."));
+			if (!error.isMember("message") ||
+				!error["message"].isString()
```
Comment (chriseth) [inline test/libsolidity/SMTCheckerJSONTest.cpp:96]: style

File: test/libsolidity/SMTCheckerJSONTest.cpp [hunk 5]
```diff
@@ -59,42 +62,59 @@ bool SMTCheckerTest::run(ostream& _stream, string const& _linePrefix, bool const
...
+			   )
+				BOOST_THROW_EXCEPTION(runtime_error("Error must have a message."));
+			if (!error.isMember("sourceLocation"))
+				continue;
+			Json::Value const& location = error["sourceLocation"];
+			if (!location.isMember("start") ||
```
Comment (chriseth) [inline test/libsolidity/SMTCheckerJSONTest.cpp]: style

File: test/libsolidity/SMTCheckerJSONTest.cpp (hunk 2 above)
Comment (leonardoalt) [reply test/libsolidity/SMTCheckerJSONTest.cpp:75]: I actually put an `if` instead, such that it doesn't crash on cases that don't lead to assertion fails.

Comment (chriseth) [reply test/libsolidity/SMTCheckerJSONTest.cpp:75]: Ah, I see!

**Review by chriseth — APPROVED** [review]

**Pull request merged**